altair/jupyter/jupyter_chart.py

```diff
@@ -241,3 +245,32 @@ def _on_change_selections(self, change):
                         selection_name, signal=value, store=store
                     ),
                 )
+
+
+def collect_transform_params(chart: TopLevelSpec) -> Set[str]:
+    """
+    Collect the names of params that are defined by transforms
+
+    Parameters
+    ----------
+    chart: Chart from which to extract transform params
+
+    Returns
+    -------
+    set of param names
+    """
+    transform_params = set()
+
+    # Handle recursive case
+    for prop in ("layer", "concat", "hconcat", "vconcat"):
+        for child in getattr(chart, prop, []):
+            transform_params.update(collect_transform_params(child))
+
+    # Handle chart's own transforms
+    transforms = getattr(chart, "transform", [])
+    transforms = transforms if transforms != alt.Undefined else []
+    for tx in transforms:
+        if isinstance(tx, alt.ExtentTransform):
```

Comment:
Instead of checking for instance `alt.ExtentTransform` could we check for `hasattr` `param`? In order to capture potential new transforms that behave similar?

Reply:
Good idea. Done in ba8b1fad1898f30c290649b01584385345fcf9e9